### PR TITLE
Add remote-viewer to check graphic password

### PR DESCRIPTION
**Pull request by cliping**
Signed-off-by: Liping Cheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:
#avocado run --vt-type libvirt graphics.functional.positive_tests.spice_only.set_passwd
JOB ID     : 91720796e6da99b59eeff7e741cac3adeda4e761
JOB LOG    : /root/avocado/job-results/job-2019-03-26T18.00-9172079/job.log
 (1/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.guest_xml: PASS (11.53 s)
 (2/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.qemu_conf: PASS (12.31 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 27.01 s

#avocado run --vt-type libvirt graphics.functional.positive_tests.vnc_only.set_passwd
JOB ID     : 83b0272d5886f0ed90a8dfca7981461163b3ccab
JOB LOG    : /root/avocado/job-results/job-2019-03-26T18.00-83b0272/job.log
 (1/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.guest_xml: PASS (11.80 s)
 (2/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.qemu_conf: PASS (12.12 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 27.18 s

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 1]
```diff
@@ -1012,6 +1019,81 @@ def block_ports(params):
...
+        fd.write(virt_viewer_conf)
+        time.sleep(1)
+
+    # run remote-viewer command
+    rv_start = threading.Thread(target=run_remote_viewer, args=(virt_viewer_file, ))
+    rv_start.start()
+    rv_start.join(3)
+
+    # stop remote-viewer command
+    cmd = "ps -ef | grep remote-viewer | grep %s" % tmpfile_prefix
```
Comment (xiaodwan) [inline libvirt/tests/src/graphics/graphics_functional.py]: pidof remote-viewer

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 2]
```diff
@@ -1012,6 +1019,80 @@ def block_ports(params):
...
+[virt-viewer]
+type=%s
+host=127.0.0.1
+port=%s
+username=root
+""" % (graphic_type, port)
+
+    # Write virt-viewer config into a tmpfile
+    tmpfile_prefix = "avocado_virt_viewer_"
+    tmp_file = tempfile.NamedTemporaryFile(prefix=(tmpfile_prefix),
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: please use data_dir.get_tmp_dir()

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 3]
```diff
@@ -1012,6 +1019,80 @@ def block_ports(params):
...
+
+    # run remote-viewer command
+    rv_start = threading.Thread(target=run_remote_viewer, args=(virt_viewer_file, ))
+    rv_start.start()
+    rv_start.join(3)
+
+    # stop remote-viewer command
+    cmd = "pidof remote-viewer"
+    res = process.run(cmd, shell=True, verbose=True)
+    utils_misc.safe_kill(res.stdout_text.strip(), signal.SIGKILL)
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: Can utils_misc.kill_process_by_pattern("remote-viewer") works for you?

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 4]
```diff
@@ -1012,6 +1019,80 @@ def block_ports(params):
...
+    if os.path.exists(virt_viewer_file):
+        os.remove(virt_viewer_file)
+
+    rv_result = q.get()
+    if graphic_type == "spice":
+        cmp_str = "main channel: opened"
+    elif graphic_type == "vnc":
+        cmp_str = "desktop resize"
+    else:
+        cmp_str = "Don't support graphic type %s." % graphic_type
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: Can we move those str to config file?

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 5]
```diff
@@ -1012,6 +1019,80 @@ def block_ports(params):
     return sockets
 
 
+def run_remote_viewer(virt_viewer_file):
+    """
+    Run remote-viewer command
+    param virt_viewer_file: remote-viewer's config file
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: ":param" and 1 line between func description and param description

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 6]
```diff
@@ -1012,6 +1019,80 @@ def block_ports(params):
...
+    q.put(res.stdout_text)
+
+
+def check_connection(graphic_type, port, graphic_passwd, test):
+    """
+    Use remote-viewer to connect guest
+    param graphic_type: graphic type
+    param port: port value
+    param graphic_passwd: graphic password
+    return: True or False
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: as above and ":return", also missing param "test"

File: libvirt/tests/cfg/graphics/graphics_functional.cfg [hunk 7]
```diff
@@ -5,6 +5,7 @@
     vnet_address = '192.168.123.1'
     macvtap_device = 'EXAMPLE.MACVTAP.DEVICE'
     bridge_device = 'EXAMPLE.BRIDGE.DEVICE'
+    remote_viewer_check = no
```
Comment (kylazhang) [inline libvirt/tests/cfg/graphics/graphics_functional.cfg]: better with '' as above and also the following settings

**Comment by cliping** [comment]
New test result:
#avocado run --vt-type libvirt graphics.functional.positive_tests.spice_only.set_passwd
JOB ID     : 8f820333d36960ceecb1d33d8b362f3030a2956b
JOB LOG    : /root/avocado/job-results/job-2019-03-29T11.54-8f82033/job.log
 (1/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.guest_xml: PASS (11.86 s)
 (2/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.qemu_conf: PASS (12.45 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 27.50 s

#avocado run --vt-type libvirt graphics.functional.positive_tests.vnc_only.set_passwd
JOB ID     : 8a18b221bccde0e18e0d07e4b8bc48952909380f
JOB LOG    : /root/avocado/job-results/job-2019-03-29T11.56-8a18b22/job.log
 (1/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.guest_xml: PASS (12.14 s)
 (2/2) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.qemu_conf: PASS (12.95 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 28.37 s

**Comment by chloerh** [comment]
LGTM

**Pull request merged**